apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V62__new_schema_snf_claims.sql

```diff
@@ -0,0 +1,843 @@
+	${logic.hsql-only}	icd_prcdr_vrsn_cd20,
+	${logic.hsql-only}	icd_prcdr_vrsn_cd21,
+	${logic.hsql-only}	icd_prcdr_vrsn_cd22,
+	${logic.hsql-only}	icd_prcdr_vrsn_cd23,
+	${logic.hsql-only}	icd_prcdr_vrsn_cd24,
+	${logic.hsql-only}	icd_prcdr_vrsn_cd25,
+	${logic.hsql-only}	prcdr_dt1,
+	${logic.hsql-only}	prcdr_dt2,
+	${logic.hsql-only}	prcdr_dt3,
+	${logic.hsql-only}	prcdr_dt4,
+	${logic.hsql-only}	prcdr_dt5,
+	${logic.hsql-only}	prcdr_dt6,
+	${logic.hsql-only}	prcdr_dt7,
+	${logic.hsql-only}	prcdr_dt8,
+	${logic.hsql-only}	prcdr_dt9,
+	${logic.hsql-only}	prcdr_dt10,
+	${logic.hsql-only}	prcdr_dt11,
+	${logic.hsql-only}	prcdr_dt12,
+	${logic.hsql-only}	prcdr_dt13,
+	${logic.hsql-only}	prcdr_dt14,
+	${logic.hsql-only}	prcdr_dt15,
+	${logic.hsql-only}	prcdr_dt16,
+	${logic.hsql-only}	prcdr_dt17,
+	${logic.hsql-only}	prcdr_dt18,
+	${logic.hsql-only}	prcdr_dt19,
+	${logic.hsql-only}	prcdr_dt20,
+	${logic.hsql-only}	prcdr_dt21,
+	${logic.hsql-only}	prcdr_dt22,
+	${logic.hsql-only}	prcdr_dt23,
+	${logic.hsql-only}	prcdr_dt24,
+	${logic.hsql-only}	prcdr_dt25 )
+-- PSQL allows us to dynamically create a table via the
+-- associated SELECT statement used to populate the table.
+${logic.psql-only} create table public.snf_claims_new as
+select
+	${logic.psql-only} cast(clm_id as bigint),
+	${logic.psql-only} cast(bene_id as bigint),
+	${logic.psql-only} cast(clm_grp_id as bigint),  
+    ${logic.hsql-only} convert(clm_id, SQL_BIGINT),
+	${logic.hsql-only} convert(bene_id, SQL_BIGINT),
+	${logic.hsql-only} convert(clm_grp_id, SQL_BIGINT),
+	last_updated,
+	clm_from_dt,
+	clm_thru_dt,
+	claim_query_code,
+	clm_admsn_dt,
+	clm_fac_type_cd,
+	clm_freq_cd,
+	clm_ip_admsn_type_cd,
+	${logic.hsql-only} clm_non_utlztn_days_cnt,
+	${logic.psql-only} cast(clm_non_utlztn_days_cnt as smallint),
+	${logic.hsql-only} clm_utlztn_day_cnt,
+	${logic.psql-only} cast(clm_utlztn_day_cnt as smallint),
+	${logic.hsql-only} bene_tot_coinsrnc_days_cnt,
+	${logic.psql-only} cast(bene_tot_coinsrnc_days_cnt as smallint),
+	clm_srvc_clsfctn_type_cd,
+	final_action,
+	${logic.hsql-only} clm_tot_chrg_amt,
+	${logic.psql-only} cast(clm_tot_chrg_amt as numeric(10,2)),
+	${logic.hsql-only} clm_pmt_amt,
+	${logic.psql-only} cast(clm_pmt_amt as numeric(10,2)),
+	clm_pps_ind_cd,
+	${logic.hsql-only} clm_pps_cptl_dsprprtnt_shr_amt,
+	${logic.psql-only} cast(clm_pps_cptl_dsprprtnt_shr_amt as numeric(10,2)),
+	${logic.hsql-only} clm_pps_cptl_excptn_amt,
+	${logic.psql-only} cast(clm_pps_cptl_excptn_amt as numeric(10,2)),
+	${logic.hsql-only} clm_pps_cptl_fsp_amt,
+	${logic.psql-only} cast(clm_pps_cptl_fsp_amt as numeric(10,2)),		
+	${logic.hsql-only} clm_pps_cptl_ime_amt,
+	${logic.psql-only} cast(clm_pps_cptl_ime_amt as numeric(10,2)),	
+	${logic.hsql-only} clm_pps_cptl_outlier_amt,
+	${logic.psql-only} cast(clm_pps_cptl_outlier_amt as numeric(10,2)),	
+	${logic.hsql-only} clm_pps_old_cptl_hld_hrmls_amt,
+	${logic.psql-only} cast(clm_pps_old_cptl_hld_hrmls_amt as numeric(10,2)),	
```

Comment:
I was not expecting all of the additional datatype changes to be a part of this. These are all new types, right?

Reply:
not really; the only thing that changes on the money fields is going from (12,2) to (10,2); minor space saving(s) on a field that would never need to support $1,111,111,111.11. I know health costs are going up but I certainly hope we don't hit a Dr's bill that hits almost a 10 billion $$$'s.

Reply:
I'm not sure that going from 12,2 to 10,2 gets us any savings. Leading zeroes will not be stored according to this: https://www.postgresql.org/docs/current/datatype-numeric.html#DATATYPE-NUMERIC-DECIMAL:

"Numeric values are physically stored without any extra leading or trailing zeroes. Thus, the declared precision and scale of a column are maximums, not fixed allocations. (In this sense the numeric type is more akin to varchar(n) than to char(n).) The actual storage requirement is two bytes for each group of four decimal digits, plus three to eight bytes overhead."

So I think this might be undesirable since it moves us further away from parity with CCW

Reply:
OK, np, will leave them (12,2)